source/exe/win32/platform_impl.cc

```diff
@@ -1,17 +1,60 @@
+#include <chrono>
+#include <thread>
+
+#include "common/buffer/buffer_impl.h"
 #include "common/common/assert.h"
 #include "common/common/thread_impl.h"
+#include "common/event/signal_impl.h"
 #include "common/filesystem/filesystem_impl.h"
 
 #include "exe/platform_impl.h"
 
 namespace Envoy {
 
+static volatile bool shutdown_pending = false;
+
+BOOL WINAPI CtrlHandler(DWORD fdwCtrlType) {
+  if (shutdown_pending) {
+    return 0;
+  }
+  shutdown_pending = true;
+
+  auto handler = Event::eventBridgeHandlersSingleton::get()[ENVOY_WIN32_SIGTERM];
+  if (!handler) {
+    return 0;
+  }
+
+  // This code is executed as part of a thread running under a Windows
```

Comment:
What do you mean by `Windows context`?

Reply:
This is one of the callbacks that are executed from a thread created and managed by the console host and not by Envoy. I bet `Windows context` is not clear enough, I will update it

Reply:
fixed in latest commit.